soc/arm/nordic_nrf/validate_enabled_instances.c

```diff
@@ -40,7 +42,9 @@
 	IF_ENABLED(CONFIG_SOC_SERIES_NRF91X, (", UARTE"#idx)) \
 	". Check nodes with status \"okay\" in zephyr.dts."
 
-#if !IS_ENABLED(CONFIG_SOC_NRF52810) && !IS_ENABLED(CONFIG_SOC_NRF52805)
+#if (!IS_ENABLED(CONFIG_SOC_NRF52810) &&	\
+	!IS_ENABLED(CONFIG_SOC_NRF52805) &&	\
+	!IS_ENABLED(CONFIG_SOC_NRF52811))
```

Comment:
For nRF52811, we need to also add the following:
```
#if IS_ENABLED(CONFIG_SOC_NRF52811)
BUILD_ASSERT(!(SPI_ENABLED(1) && I2C_ENABLED(0)),
	     "Only one of the following peripherals can be enabled: "
	     "SPI1, SPIM1, SPIS1, TWI0, TWIM0, TWIS0. "
	     "Check nodes with status \"okay\" in zephyr.dts.");
#endif
```

Reply:
> For nRF52811, we need to also add the following:

Added, and tagged on a DTSI fix for this SoC I noticed while looking at it for this PR.